xds/src/main/java/io/grpc/xds/XdsServerBuilder.java

```diff
@@ -41,10 +45,22 @@
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/7514")
 public final class XdsServerBuilder extends ForwardingServerBuilder<XdsServerBuilder> {
 
+  public static final Attributes.Key<AtomicReference<FilterChainSelector>>
+          ATTR_FILTER_CHAIN_SELECTOR_REF = Attributes.Key.create(
+          "io.grpc.xds.XdsServerBuilder.filterChainSelectorRef");
+  private static final long RETRY_DELAY_NANOS = TimeUnit.MINUTES.toNanos(1);
+
   private final NettyServerBuilder delegate;
   private final int port;
   private XdsServingStatusListener xdsServingStatusListener;
   private AtomicBoolean isServerBuilt = new AtomicBoolean(false);
+  private final AtomicReference<FilterChainSelector> filterChainSelectorRef =
+          new AtomicReference<>();
+  private XdsClientPoolFactory xdsClientPoolFactory =
+          SharedXdsClientPoolProvider.getDefaultProvider();
+  private final ScheduledExecutorService timeService =
+          SharedResourceHolder.get(GrpcUtil.TIMER_SERVICE);
```

Comment:
This is acquiring a reference count; it has to be pared with a `release()`. I think the resource (or a ObjectPool) needs to be passed down to XdsServerWrapper and have XdsServerWrapper allocate it on first start().